src/test/java/de/tum/in/www1/artemis/iris/PyrisLectureIngestionTest.java

```diff
@@ -0,0 +1,308 @@
+package de.tum.in.www1.artemis.iris;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.util.LinkedMultiValueMap;
+
+import de.tum.in.www1.artemis.course.CourseUtilService;
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.Lecture;
+import de.tum.in.www1.artemis.domain.iris.settings.IrisCourseSettings;
+import de.tum.in.www1.artemis.domain.lecture.AttachmentUnit;
+import de.tum.in.www1.artemis.lecture.LectureUtilService;
+import de.tum.in.www1.artemis.repository.CourseRepository;
+import de.tum.in.www1.artemis.repository.LectureRepository;
+import de.tum.in.www1.artemis.repository.iris.IrisSettingsRepository;
+import de.tum.in.www1.artemis.service.connectors.pyris.PyrisJobService;
+import de.tum.in.www1.artemis.service.connectors.pyris.PyrisStatusUpdateService;
+import de.tum.in.www1.artemis.service.connectors.pyris.PyrisWebhookService;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.lectureingestionwebhook.PyrisLectureIngestionStatusUpdateDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.status.PyrisStageDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.status.PyrisStageState;
+import de.tum.in.www1.artemis.user.UserUtilService;
+
+class PyrisLectureIngestionTest extends AbstractIrisIntegrationTest {
```

Comment:
These tests require the dependencies from the AbstractIrisIntegrationTest, that's why they are in that class. The integration tests with upload lectures for example is not possible because the ingestion is like a side effect and there is nothing returned or saved on the Artemis side. More integration test / refractoring will happen when we start saving the Pyris Ingestion state, that way the integration tests can be done seamlessly.